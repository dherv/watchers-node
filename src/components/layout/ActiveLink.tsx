import { useRouter } from "next/router";
import styled from "styled-components";

const ActiveLink = ({ children, href }) => {
  const router = useRouter();

  const handleClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
    event.preventDefault();
    router.push(href);
  };

  return (
    <ListItem href={href} path={router.pathname}>
      <Anchor href={href} onClick={handleClick} height={60}>
        {children}
      </Anchor>
    </ListItem>
  );
};

const Anchor = styled.a<{ href: string; height: number }>`
  height: ${props => props.height}px;
  line-height: ${props => props.height}px;
  width: 100%;
`;

const ListItem = styled.li<{ path: string; href: string }>`
  font-weight: 100;
  font-size: 1.1rem;
  margin: 0 4rem;
  position: relative;
  &::after {
    content: "";
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    border-width: 2px;
    border-style: solid;
    border-color: ${props =>
      props.path === props.href ? "#2e8e89" : "transparent"};
  }
`;
export { Anchor, ListItem };
export default ActiveLink;
