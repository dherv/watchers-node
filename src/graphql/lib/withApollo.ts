import withApollo from "next-with-apollo";
import ApolloClient, { InMemoryCache } from "apollo-boost";
// import { GRAPHQL_URL } from '../configs';

export default withApollo(
  ({ initialState }) =>
    new ApolloClient({
      uri: "http://localhost:3000/graphql",
      cache: new InMemoryCache().restore(initialState || {})
    })
);
