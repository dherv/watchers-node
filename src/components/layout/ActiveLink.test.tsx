import { shallow, mount } from "enzyme";
import React from "react";
import ActiveLink from "./ActiveLink";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import { NextRouter } from "next/router";

describe("ActiveLink component", () => {
  const props = {
    href: "/movies",
    children: "movies"
  };
  const router = ({
    pathname: "/users/$user",
    route: "/users/$user",
    query: { user: "nikita" },
    asPath: "/users/nikita",
    push: jest.fn()
  } as unknown) as NextRouter;

  const wrapper = mount(
    <RouterContext.Provider value={router}>
      <ActiveLink {...props} />
    </RouterContext.Provider>
  );

  describe("UI", () => {
    test("should display an anchor", () => {
      expect.assertions(1);
      expect(wrapper.find("a")).toHaveLength(1);
    });

    test("should use the props correctly within the anchor tag", () => {
      expect.assertions(2);
      expect(wrapper.find("a").prop("href")).toEqual(props.href);
      expect(wrapper.find("a").text()).toEqual(props.children);
    });
  });

  describe("Events", () => {
    test("should call router.push on click", () => {
      expect.assertions(1);
      const link = wrapper.find("a").first();
      link.simulate("click");
      expect(router.push).toHaveBeenCalledWith(props.href);
    });
  });
});
