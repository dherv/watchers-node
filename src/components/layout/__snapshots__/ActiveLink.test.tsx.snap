// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveLink component snapshot should match 1`] = `
.c1 {
  height: 60px;
  line-height: 60px;
  width: 100%;
}

.c0 {
  font-weight: 100;
  font-size: 1.1rem;
  margin: 0 4rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-width: 2px;
  border-style: solid;
  border-color: #2e8e89;
}

<ActiveLink
  href="/movies"
>
  <ActiveLink__ListItem
    href="/movies"
    path="/movies"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "ActiveLink__ListItem-sc-1si0yf-1",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "font-weight:100;font-size:1.1rem;margin:0 4rem;position:relative;&::after{content:\\"\\";position:absolute;bottom:0;left:0;right:0;border-width:2px;border-style:solid;border-color:",
              [Function],
              ";}",
            ],
          },
          "displayName": "ActiveLink__ListItem",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "ActiveLink__ListItem-sc-1si0yf-1",
          "target": "li",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      href="/movies"
      path="/movies"
    >
      <li
        className="c0"
        href="/movies"
      >
        <ActiveLink__Anchor
          height={60}
          href="/movies"
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ActiveLink__Anchor-sc-1si0yf-0",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "height:",
                    [Function],
                    "px;line-height:",
                    [Function],
                    "px;width:100%;",
                  ],
                },
                "displayName": "ActiveLink__Anchor",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ActiveLink__Anchor-sc-1si0yf-0",
                "target": "a",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height={60}
            href="/movies"
            onClick={[Function]}
          >
            <a
              className="c1"
              height={60}
              href="/movies"
              onClick={[Function]}
            >
              movies
            </a>
          </StyledComponent>
        </ActiveLink__Anchor>
      </li>
    </StyledComponent>
  </ActiveLink__ListItem>
</ActiveLink>
`;
